dspy/teleprompt/teleprompt_optuna.py

```diff
@@ -36,7 +36,7 @@ def objective(self, trial):
         for (name, compiled_predictor), (_, program2_predictor) in zip(self.compiled_teleprompter.named_predictors(), program2.named_predictors()):
             all_demos = compiled_predictor.demos
             demo_index = trial.suggest_int(f"demo_index_for_{name}", 0, len(all_demos) - 1)
-            selected_demo = dict(all_demos[demo_index])
+            selected_demo = all_demos[demo_index]
```

Comment:
If this causes a real problem I would rather go with an additional line below of type:

```python
if not isinstances(selected_demo, dict):
   selected_demo = dict(selected_demo)
```
what do you think ?